pkg/controller/packetcapture/packetcapture_controller.go

```diff
@@ -0,0 +1,344 @@
+			return reconcile.Result{}, err
+		}
+		packetcaptureapi.Status.Conditions = status.UpdateStatusCondition(packetcaptureapi.Status.Conditions, ts.Status.Conditions)
+		if err := r.client.Status().Update(ctx, packetcaptureapi); err != nil {
+			log.WithValues("reason", err).Info("Failed to create packetcapture status conditions.")
+			return reconcile.Result{}, err
+		}
+	}
+
+	variant, installationSpec, err := utils.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Installation not found", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if variant != operatorv1.TigeraSecureEnterprise {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, fmt.Sprintf("Waiting for network to be %s", operatorv1.TigeraSecureEnterprise), nil, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	managementCluster, err := utils.GetManagementCluster(ctx, r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading ManagementCluster", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Packet capture is disabled in multi tenant management cluster
+	if r.multiTenant && managementCluster != nil {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Packet capture is disabled in the multitenant management cluster.", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	if !utils.IsAPIServerReady(r.client, reqLogger) {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Tigera API server to be ready", nil, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Validate that the tier watch is ready before querying the tier to ensure we utilize the cache.
+	if !r.tierWatchReady.IsReady() {
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for Tier watch to be established", err, reqLogger)
+		return reconcile.Result{RequeueAfter: utils.StandardRetry}, nil
+	}
+
+	// Ensure the allow-tigera tier exists, before rendering any network policies within it.
+	if err := r.client.Get(ctx, client.ObjectKey{Name: networkpolicy.TigeraComponentTierName}, &v3.Tier{}); err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Waiting for allow-tigera tier to be created, see the 'tiers' TigeraStatus for more information", err, reqLogger)
+			return reconcile.Result{RequeueAfter: utils.StandardRetry}, nil
+		}
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Error querying allow-tigera tier", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	// Create a component handler to manage the rendered component.
+	handler := utils.NewComponentHandler(log, r.client, r.scheme, packetcaptureapi)
+
+	// Render the desired objects from the CRD and create or update them.
+	reqLogger.V(3).Info("rendering components")
```

Comment:
We can get rid of this log, there are logs in the handler code that tell us this instead.